cloud/smartos/vmadm

```diff
@@ -0,0 +1,250 @@
+       SmartOS system. All 3 of: OS Virtual Machines (SmartOS zones), LX
+       Virtual Machines and KVM Virtual Machines can be managed. vmadm allows
+       you to create, inspect, modify and delete virtual machines on the local
+       system.
+
+options:
+    state:
+        required: true
+        choices: [present, absent, rebooted, started, stopped, updated]
+        description:
+        - which state the virtual machine should be in
+    force:
+        required: false
+        choices: [yes, no, true, false, 1, 0, True, False]
+        description:
+        - if true, will immediately shutdown/reboot, instead of giving the vm time do so, cleanly
+    uuid:
+        required: false
+        description:
+        - virtual machine uuid (either this or alias)
+    alias:
+        required: false
+        description:
+        - virtual machine alias (either this or uuid)
+    manifest:
+        required: false
+        description:
+        - a valid json file for create/update
+    proplist:
+        required: false
+        description:
+        - a k=v whitespace-separated list to pass to vmadm update
+'''
+
+EXAMPLES = '''
+# Create a vm
+- vmadm: manifest=/opt/custom/zones/webserver.json state=present
+
+# Start a previously created vm (by alias)
+- vmadm: alias=webserver state=started
+
+# Start a previously created vm (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=started
+
+# Stop a previously created vm (by alias)
+- vmadm: alias=webserver state=stopped
+
+# Forcibly stop a previously created vm (by alias)
+- vmadm: alias=webserver state=stopped force=yes
+
+# Stop a previously created vm (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=stopped
+
+# Forcibly stop a previously created vm (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=stopped force=yes
+
+# Reboot a vm (by alias)
+- vmadm: alias=webserver state=rebooted
+
+# Forcibly reboot a vm (by alias)
+- vmadm: alias=webserver state=rebooted force=yes
+
+# Reboot a vm (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=rebooted
+
+# Forcibly reboot a vm (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=rebooted force=yes
+
+# Update a vm, using a property list (by alias)
+- vmadm: alias=webserver proplist='ram=256 quota=64' state=updated
+
+# Update a vm, using a property list (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 proplist='ram=256 quota=64' state=updated
+
+# Update a vm, using payload from a json file (by alias)
+- vmadm: alias=webserver manifest=/tmp/update.json state=updated
+
+# Update a vm, using payload from a json file (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 manifest=/tmp/update.json state=updated
+
+# Delete a VM (by alias)
+- vmadm: alias=webserver state=absent
+
+# Delete a VM (by uuid)
+- vmadm: uuid=e19429c2-f412-4210-b0cf-5ed59b81b8f1 state=absent
+'''
+
+import json
```

Comment:
not all machines will have json, some might have simplejson, look at other files for conditional imports